app/src/organisms/Desktop/ProtocolVisualization/LabwareSlotContainer.tsx

```diff
@@ -0,0 +1,209 @@
+  const labwareDisplayName = labwareDef.metadata.displayName
+
+  const liquidDisplayColors = Object.fromEntries(
+    liquids.map(liquid => [liquid.id, liquid.displayColor ?? COLORS.grey40])
+  )
+  const allWellContentsForActiveItem = getAllWellContentsAtFrame(
+    robotState.liquidState,
+    labwareDef
+  )
+  const wellContents =
+    allWellContentsForActiveItem != null
+      ? allWellContentsForActiveItem[topLabwareOnSlotId]
+      : null
+
+  const wellFill = wellFillFromWellContents(wellContents, liquidDisplayColors)
+  const activeWellName =
+    pipetteTemporalProperties != null
+      ? pipetteTemporalProperties[1].wellName
+      : null
+  const wellGroup: WellGroup | null =
+    activeWellName != null
+      ? {
+          [activeWellName]: null,
+        }
+      : null
+  const { wells } = labwareDef
+
+  const pipetteLocationLiquidState =
+    pipetteTemporalProperties != null
+      ? robotState.liquidState.pipettes[pipetteTemporalProperties[0]]?.[0]
+      : null
+  const labwareLocationLiquidState =
+    activeWellName != null
+      ? robotState.liquidState.labware[topLabwareOnSlotId]?.[activeWellName]
+      : null
+
+  const tipMaxVolume =
+    pipetteTemporalProperties != null
+      ? pipetteEntities[pipetteTemporalProperties[0]].spec.liquids.default
+          .maxVolume
+      : 0
+
+  const labwareViewBox = getLabwareViewBox(labwareDef)
+  const ingredNames = liquids.reduce(
+    (acc: Record<string, string>, { id, displayName }) => {
+      acc[id] = displayName
+      return acc
+    },
+    {}
+  )
+
+  return (
+    <>
+      {activeWellName != null ? (
+        <WellContainer
+          wells={wells}
+          params={params}
+          activeWellName={activeWellName}
+          wellColor={wellFill[activeWellName]}
+          labwareLocationLiquidState={labwareLocationLiquidState}
+          pipetteLocationLiquidState={pipetteLocationLiquidState}
+          liquids={liquids}
+          tipMaxVolume={tipMaxVolume}
+        />
+      ) : null}
+      <div className={styles.container}>
+        <div className={styles.header}>
+          <Tag text={t('labware')} type="default" shrinkToContent />
+          <div className={styles.info_label_container}>
+            {labware[topLabwareOnSlotId]?.stack
+              .filter(item => item !== topLabwareOnSlotId)
+              .reverse()
+              .map(item => {
+                if (moduleEntities[item] != null) {
+                  return (
+                    <RobotInfoLabel
+                      key={item}
+                      iconName={
+                        MODULE_ICON_NAME_BY_TYPE[moduleEntities[item].type]
+                      }
+                    />
+                  )
+                } else if (labware[item] != null) {
+                  return <RobotInfoLabel key={item} iconName="stacked" />
+                } else {
+                  return <RobotInfoLabel key={item} deckLabel={slot} />
+                }
+              })}
```

Comment:
maybe for readability
```suggestion
            {labware[topLabwareOnSlotId]?.stack
              .reverse()
              .reduce<JSX.Element[]>((acc, item) => {
                if (item !== topLabwareOnSlotId) {
                  if (moduleEntities[item] != null) {
                    acc.push(
                      <RobotInfoLabel
                        key={item}
                        iconName={
                          MODULE_ICON_NAME_BY_TYPE[moduleEntities[item].type]
                        }
                      />
                    )
                  } else if (labware[item] != null) {
                    acc.push(<RobotInfoLabel key={item} iconName="stacked" />)
                  } else {
                    acc.push(<RobotInfoLabel key={item} deckLabel={slot} />)
                  }
                }
                return acc
              }, [] as JSX.Element[])}
          </div>
```